### ISSUE TITLE
different php version return different format

### ISSUE THREAD

**Issue by genepeng** [giggsey/libphonenumber-for-php#234]
I want to format (+86) 18616177661, 
PHP 7.0.10-1~dotdeb+8.1       formatted as +86 18616177661
PHP 7.0.22-0ubuntu0.17.04.1 formatted as  +86 186 1617 7661

below are my test commands

` root@00636eb6be20:/var/osl# php -v
PHP 7.0.10-1~dotdeb+8.1 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.10-1~dotdeb+8.1, Copyright (c) 1999-2016, by Zend Technologies
root@00636eb6be20:/var/osl# php test.php

+86 18616177661

root@00636eb6be20:/var/osl# exit
exit
➜  microapp-student-portal git:(ZEN-2981) ✗ php test.php

+86 186 1617 7661

➜  microapp-student-portal git:(ZEN-2981) ✗ php -v
PHP 7.0.22-0ubuntu0.17.04.1 (cli) (built: Aug  8 2017 22:03:30) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.22-0ubuntu0.17.04.1, Copyright (c) 1999-2017, by Zend Technologies `

I am using 
`            "name": "giggsey/libphonenumber-for-php",
            "version": "8.8.10",`

my php code in test.php

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$phoneNumber = '(+86) 18616177661';
$phoneUtil = \libphonenumber\PhoneNumberUtil::getInstance();
try {
    $parsedPhoneNumber = $phoneUtil->parse($phoneNumber, null);
    echo $phoneUtil->format($parsedPhoneNumber, \libphonenumber\PhoneNumberFormat::INTERNATIONAL);
    echo "\n";
} catch (\libphonenumber\NumberParseException $e) {
            echo "Failed to format $phoneNumber:" . $e->getMessage();
}
```

**Comment by giggsey** [comment]
Confirmed, although I seem to get the opposite result:

Using the latest libphonenumber-for-php, and docker images for the different PHP versions:

    $ docker run -v ~/lp:/opt php:7.0.10 php /opt/test.php
    +86 186 1617 7661

    $ docker run -v ~/lp:/opt php:7.0.22 php /opt/test.php
    +86 18616177661

Until I can investigate properly, I think it'll be something with the PCRE version included in PHP.

**Comment by giggsey** [comment]
Disabling pcre.jit avoids the problem:

    $ docker run -v ~/lp:/opt php:7.0.10 php -d pcre.jit=0 /opt/test.php
    +86 186 1617 7661

    $ docker run -v ~/lp:/opt php:7.0.22 php -d pcre.jit=0 /opt/test.php
    +86 186 1617 7661

**Comment by genepeng** [comment]
@giggsey my PHP 7.0.10 running in docker, and my PHP 7.0.22 running on my local machine, 

I used below command to install php in docker
`apt-get install -y --no-install-recommends \
      nginx \
      php7.0-cli=7.0.10-1~dotdeb+8.1 \
      php7.0-common=7.0.10-1~dotdeb+8.1 \
      php7.0-curl=7.0.10-1~dotdeb+8.1 \
      php7.0-fpm=7.0.10-1~dotdeb+8.1 \
      php7.0-intl=7.0.10-1~dotdeb+8.1 \
      php7.0-json=7.0.10-1~dotdeb+8.1 \
      php7.0-mbstring=7.0.10-1~dotdeb+8.1 \
      php7.0-mcrypt=7.0.10-1~dotdeb+8.1 \
      php7.0-mysql=7.0.10-1~dotdeb+8.1 \
      php7.0-opcache=7.0.10-1~dotdeb+8.1 \
      php7.0-readline=7.0.10-1~dotdeb+8.1 \
      php7.0-xml=7.0.10-1~dotdeb+8.1 \
      php7.0=7.0.10-1~dotdeb+8.1`

it worked before, but the format changed recently (at the beginning of  2018 Feb)

**Comment by genepeng** [comment]
@giggsey it works on my local machine with **pcre.jit=1**, but have to set **pcre.jit=0** in docker

**Comment by giggsey** [comment]
Related issues: https://github.com/giggsey/libphonenumber-for-php/issues/79 and maybe https://bugs.php.net/bug.php?id=70304

**Comment by giggsey** [comment]
Looks like it needs a similar patch to #89.

Problem regex:
https://github.com/giggsey/libphonenumber-for-php/blob/master/src/data/PhoneNumberMetadata_CN.php#L451

> 1(?:[3-57-9]|66)

Splitting it up to:

> 1(?:(?:[3-5]|[7-9])|66)

means it works on both versions of PHP7 with jit enabled.

However, this is not a scalable solution in the future, so another solution is required.

**Comment by giggsey** [comment]
I updated one of the previous comments with the PCRE versions found in both docker images. Annoyingly, the newer PHP version has an older PCRE version.

As commented in the PHP bug above, PHP 7.0.3 was released, and bundles in PCRE version 8.38. This fixed the PCRE bug with JIT, and means the regex used throughout this project will work correctly.

Note: this isn't a problem before PHP 7, as the JIT functionality did not exist.

This was fixed in the Docker images as part of https://github.com/docker-library/php/issues/495, although I'm not sure why versions after 7.0.17 downgraded the PCRE version:

> $ docker run -v ~/lp:/opt php:7.0.17 php -d pcre.jit=1 -r "echo PHP_VERSION . PHP_EOL; echo PCRE_VERSION;"
7.0.17
8.38 2015-11-23

> $ docker run -v ~/lp:/opt php:7.0.18 php -d pcre.jit=1 -r "echo PHP_VERSION . PHP_EOL; echo PCRE_VERSION;"
7.0.18
8.35 2014-04-04

If there are problems with non docker PHP versions, then I believe that's a problem with how they are being packaged.
